processing/src/main/java/org/apache/druid/segment/nested/NestedFieldLiteralColumnIndexSupplier.java

```diff
@@ -574,22 +579,34 @@ public BitmapColumnIndex forValue(@Nullable String value)
         public double estimateSelectivity(int totalRows)
         {
           if (longValue == null) {
-            return (double) getBitmap(localDictionary.indexOf(0)).size() / totalRows;
+            if (inputNull) {
+              return (double) getBitmap(localDictionary.indexOf(0)).size() / totalRows;
```

Comment:
This is one of those hard-codings of null as 0.  I wonder if maybe we should add an `indexOfNull()` method that would only be implemented by the things that actually have null as 0 as an extra safe-guard to ensure that we only do that when we know that we are using null.  The local implementation can special case `0` more safely.

What do you think?

Reply:
right now the global dictionary id is always the null value, its the local dictionary and bitmaps that you cant assume this. Since we are looking for indexOf 0 in the local dictionary, this will give us an empty bitmap if null isn't present in the column

Where would this method live?

Reply:
Only exist on the local dictionary objects?